lib/backend-api/src/query.rs

```diff
@@ -24,10 +25,21 @@ use crate::{
 /// the API, and should not be used where possible.
 pub async fn fetch_webc_package(
     client: &WasmerClient,
-    ident: &WebcIdent,
+    ident: &PackageIdent,
     default_registry: &Url,
 ) -> Result<webc::compat::Container, anyhow::Error> {
-    let url = ident.build_download_url_with_default_registry(default_registry);
+    let url = match ident {
+        PackageIdent::Named(n) => Url::parse(&format!(
+            "{default_registry}/{}:{}",
```

Comment:
`Url::to_string()` ends with a trailing slash, so this would have double slashes.
We could just clone the URL and use `Url::set_path()`.

But also, note that `NamedPackageIdent` has a `registry` field.

Also, this method needs an optional auth token, because unnamed private packages won't be downloadable otherwise, but we can do that as a followup.

But since this isn't actually actively used anywhere, we can do all of that in a followup to not block the merge.
=> create followup issue, @xdoardo

Reply:
Ignore the `auth token` part, that's already on the client of course, I thought this was a standalone method.

Reply:
Tracked in https://github.com/wasmerio/wasmer/issues/4598